infrastructure/src/main/java/org/corfudb/infrastructure/logreplication/infrastructure/SessionManager.java

```diff
@@ -99,24 +99,13 @@ public class SessionManager {
      * @param upgradeManager      upgrade management module
      */
     public SessionManager(@Nonnull TopologyDescriptor topology, CorfuRuntime corfuRuntime,
-                          ServerContext serverContext, LogReplicationUpgradeManager upgradeManager) {
+                          ServerContext serverContext, LogReplicationUpgradeManager upgradeManager,
+                          String localCorfuEndpoint) {
         this.topology = topology;
         this.runtime = corfuRuntime;
         this.corfuStore = new CorfuStore(corfuRuntime);
 
-        // serverContext.getLocalEndpoint currently provides a string with address = localhost and port = LR server
-        // port.  But the endpoint needed here is with the Corfu port, i.e., 9000.  So create a NodeLocator
-        // using the endpoint available from serverContext first and then create a new NodeLocator which replaces the
-        // port with the Corfu port.
-        // TODO: Once IPv6 changes are available, the 2nd NodeLocator need not be created.  A utility method which
-        //  generates the endpoint given an IP address and port - getVersionFormattedHostAddressWithPort() - should be
-        //  used
-        NodeLocator nodeLocator = NodeLocator.parseString(serverContext.getLocalEndpoint());
-        NodeLocator lrNodeLocator = NodeLocator.builder().host(nodeLocator.getHost())
-                .port(topology.getLocalClusterDescriptor().getCorfuPort())
-                .build();
-
-        this.localCorfuEndpoint = lrNodeLocator.toEndpointUrl();
+        this.localCorfuEndpoint = localCorfuEndpoint;
```

Comment:
Is this going to be used only for testing, right? If yes, please update the variable description.

Reply:
No, its not used only for testing..its used in other places in the code where we connect to corfu